zebra-rpc/src/methods/get_block_template_rpcs.rs

```diff
@@ -529,6 +566,56 @@ where
         }
         .boxed()
     }
+
+    fn get_mining_info(&self) -> BoxFuture<Result<types::get_mining_info::Response>> {
+        let network = self.network;
+        let solution_rate_fut = self.get_network_sol_ps(None, None);
+        async move {
+            Ok(types::get_mining_info::Response::new(
+                network,
+                solution_rate_fut.await?,
+            ))
+        }
+        .boxed()
+    }
+
+    fn get_network_sol_ps(
+        &self,
+        num_blocks: Option<usize>,
+        height: Option<i32>,
+    ) -> BoxFuture<Result<u128>> {
+        let num_blocks = num_blocks
+            .map(|num_blocks| num_blocks.max(1))
+            .unwrap_or(DEFAULT_SOLUTION_RATE_WINDOW_SIZE);
+        let height = height.and_then(|height| (height > 1).then_some(Height(height as u32)));
+        let mut state = self.state.clone();
+
+        async move {
+            let request = ReadRequest::SolutionRate { num_blocks, height };
+
+            let response = state
+                .ready()
+                .and_then(|service| service.call(request))
+                .await
+                .map_err(|error| Error {
+                    code: ErrorCode::ServerError(0),
+                    message: error.to_string(),
+                    data: None,
+                })?;
+
+            let solution_rate = match response {
+                ReadResponse::SolutionRate(solution_rate) => solution_rate.ok_or(Error {
+                    code: ErrorCode::ServerError(0),
```

Comment:
Does this need to be error -10 to match the other "no blocks in state" errors?

Reply:
It looks like the other "no blocks in state" errors are all `ErrorCode::ServerError(0)`.